### DISCUSSION TITLE
Event to limit types of fields in a  Field Layout

### DISCUSSION THREAD

**Discussion by janhenckens** [craftcms/cms#11634] [Ideas] [answered]
I'm building out a proof of concept for a new plugin and I would like to be able to limit the fields (or the types of fields) that the user can select.

The plugin would make it possible to define custom properties on a button, and I'd like to do so using custom fields the user can create. I would however like the avoid fields like Matrix and Super Table being used here, as that's not intended use case (of either). 

An event that would hook into [getAvailableCustomFields](https://github.com/craftcms/cms/blob/develop/src/models/FieldLayout.php#L332) for example.

Happy to try my hand at a PR for this is if it's something worth considering :)

**Comment by brandonkelly** [comment, marked as answer]
Just added a new `EVENT_DEFINE_CUSTOM_FIELDS` event for Craft 4.2! (a80dfc6d16ca0d313499151c9709bd64cc3980ed)

**Reply by brandonkelly** [reply]
Craft 4.2.0 is out now with that new event ✨

**Reply by janhenckens** [reply]
That does exactly what I had in mind, and I have it working in the plugin. Cheers!
